src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+    /* default */ private static final double                           GRAVITY_MULTIPLIER    = 0.20000000298023224D;
+    /* default */ private static final float                            ARROW_VELOCITY        = 1.6F;
+    /* default */ private static final float                            SOUND_MULTIPLIER      = 0.4F;
+    /* default */ private static final float                            SOUND_ADDER           = 0.8F;
+    /* default */ private static final float                            ENTITY_EYE_HEIGHT     = 1.74F;
+
+    private final EntityAIAttackMelee aiAttackOnCollide = new EntityAIAttackMelee(this, 1.2D, false)
+    {
+        /**
+         * Resets the task
+         */
+        public void resetTask()
+        {
+            super.resetTask();
+            AbstractArcherBarbarian.this.setSwingingArms(false);
+        }
+
+        /**
+         * Execute a one shot task or start executing a continuous task
+         */
+        public void startExecuting()
+        {
+            super.startExecuting();
+            AbstractArcherBarbarian.this.setSwingingArms(true);
+        }
+    };
+
+    public AbstractArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.setSize(ENTITY_WIDTH, ENTITY_HEIGHT);
+        this.setCombatTask();
+    }
+
+    protected void initEntityAI()
+    {
+        this.tasks.addTask(1, new EntityAISwimming(this));
+        this.tasks.addTask(2, new EntityAIRestrictSun(this));
+        this.tasks.addTask(3, new EntityAIAvoidEntity(this, EntityWolf.class, 6.0F, 1.0D, 1.2D));
+        this.tasks.addTask(5, new EntityAIWalkToRandomHuts(this, 2.0D));
+        this.tasks.addTask(6, new EntityAIWatchClosest(this, EntityPlayer.class, 8.0F));
+        this.tasks.addTask(6, new EntityAILookIdle(this));
+        this.targetTasks.addTask(1, new EntityAIHurtByTarget(this, false, new Class[0]));
+        this.targetTasks.addTask(2, new EntityAINearestAttackableTarget(this, EntityPlayer.class, true));
+        this.targetTasks.addTask(3, new EntityAINearestAttackableTarget(this, EntityCitizen.class, true));
+    }
+
+    @Override
+    protected void applyEntityAttributes()
+    {
+        super.applyEntityAttributes();
+        this.getEntityAttribute(SharedMonsterAttributes.FOLLOW_RANGE).setBaseValue(35.0D);
+        this.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED).setBaseValue(0.2D);
+        this.getEntityAttribute(SharedMonsterAttributes.ATTACK_DAMAGE).setBaseValue(1.0D);
+        this.getEntityAttribute(SharedMonsterAttributes.ARMOR).setBaseValue(2.0D);
+        this.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).setBaseValue(this.getHealthBasedOnRaidLevel());
+    }
+
+    protected double getHealthBasedOnRaidLevel()
+    {
+        if (colony != null)
+        {
+            final int raidLevel = (int) (colony.getRaidLevel() * 1.5);
+            return BARBARIAN_BASE_HEALTH + raidLevel;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this addition operation to a "double". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2184)